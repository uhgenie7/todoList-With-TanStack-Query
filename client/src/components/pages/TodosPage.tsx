import TodoItemCreateForm from '@src/components/molecules/TodoItemCreateForm';
import TodoList from '@src/components/molecules/TodoList';
import styled from 'styled-components';
import ExtendsAsyncBoundary from '../boundaries/ExtendsAsyncBoundary';

const TodosPage = () => {
  return (
    <Container>
      <div className="row">
        <section>
          <TodoItemCreateForm />
          <ExtendsAsyncBoundary>
            <TodoList />
          </ExtendsAsyncBoundary>
        </section>
      </div>
    </Container>
  );
};

export default TodosPage;

const Container = styled.div`
  .row {
    display: flex;
  }
`;
