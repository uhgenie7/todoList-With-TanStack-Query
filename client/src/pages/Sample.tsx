export const Sample = () => {
  return (
    <main>
      <h1>test jest</h1>
      <p>ggg</p>
    </main>
  );
};
