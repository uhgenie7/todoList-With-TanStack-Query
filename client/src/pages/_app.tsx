import { useState } from 'react';
import type { AppProps } from 'next/app';
import { Hydrate, QueryClient, QueryClientProvider } from '@tanstack/react-query';
import { RecoilRoot } from 'recoil';
import AppLayout from '@src/components/atoms/AppLayout';
import GlobalStyle from '@src/styles/globalStyle';

function MyApp({ Component, pageProps }: AppProps) {
  const [queryClient] = useState(
    new QueryClient({
      defaultOptions: {
        queries: {
          refetchOnWindowFocus: false,
          refetchOnMount: false,
          retry: false,
        },
      },
    }),
  );

  return (
    <QueryClientProvider client={queryClient}>
      <RecoilRoot>
        <Hydrate state={pageProps.dehydratedState}>
          <GlobalStyle />
          <AppLayout>
            <Component {...pageProps} />
          </AppLayout>
        </Hydrate>
      </RecoilRoot>
    </QueryClientProvider>
  );
}

export default MyApp;
