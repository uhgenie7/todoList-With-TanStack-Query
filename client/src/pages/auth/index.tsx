import { useEffect } from 'react';
import Button from '@src/components/atoms/Button';
import AuthForm from '@src/components/organism/AuthForm';
import { useRouter } from 'next/router';
import { loginAPI } from '@src/apis/auth';
import styled from 'styled-components';
import { QueryUserAuthKeys } from '@src/constants/QueryUserAuthKeys';

const Auth = () => {
  const router = useRouter();
  const handleNavigateSignUp = () => router.push('/auth/signup');

  return (
    <Container>
      <h2>로그인</h2>
      <div className="center">
        <AuthForm queryKey={QueryUserAuthKeys.login} buttonValue="로그인" handleLoginAPI={loginAPI} />
        <Button isCorrect={true} buttonValue="회원가입" onClick={handleNavigateSignUp} />
      </div>
    </Container>
  );
};

export default Auth;

const Container = styled.div`
  .center {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
  }
`;
