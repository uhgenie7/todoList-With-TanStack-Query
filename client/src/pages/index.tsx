import Head from 'next/head';
import Todos from '@src/components/organism/Todos';
import { Suspense } from 'react';
import Header from '@src/components/atoms/Header';
const Home = () => {
  return (
    <div>
      <Head>
        <title>wanted-pre-onboarding-challenge-fe-1</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <Header />
      <Suspense fallback={<div>loading</div>}>
        <Todos />
      </Suspense>
    </div>
  );
};

export default Home;

// export const getServerSideProps: GetServerSideProps = async () => {
//   const { data: todos } = await getTodoListAPI();

//   return {
//     props: { todos },
//   };
// };
