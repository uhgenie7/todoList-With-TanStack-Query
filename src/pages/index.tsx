import { useEffect } from 'react';
import type { NextPage } from 'next';
import Head from 'next/head';
import { useRouter } from 'next/router';
import showToast from '@src/libs/common';
import { Suspense } from 'react';
import AppLayout from '@src/components/atoms/AppLayout';

const Home: NextPage = () => {
  const toast = showToast();
  const router = useRouter();
  useEffect(() => {
    if (!localStorage.getItem('token')) {
      toast.success('로그인 먼저 해주세요');
      router.push('/auth');
    }
  }, []);
  return (
    <div>
      <Head>
        <title>wanted-pre-onboarding-challenge-fe-1</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>
      <AppLayout>
        <Suspense fallback={`loading`}>
          <h1>Home</h1>
          <p>여기는 루트입니다</p>
        </Suspense>
      </AppLayout>
    </div>
  );
};

export default Home;
