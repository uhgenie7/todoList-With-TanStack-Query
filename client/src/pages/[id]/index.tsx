import { useRouter } from 'next/router';
import TodosPage from '@src/components/pages/TodosPage';
import TodoById from '@src/components/molecules/TodoById';
import AsyncBoundary from '@src/components/boundaries/AsyncBoundary';
import DefaultErrorFallback from '@src/components/atoms/DefaultErrorFallback';
import PageLoader from '@src/components/atoms/PageLoader';
import styled from 'styled-components';

const TodoDetail = () => {
  const {
    query: { id },
  } = useRouter();

  return (
    <Container>
      {id && (
        <>
          <TodosPage />
          <div className="todoByIdSection">
            <AsyncBoundary rejectedFallback={DefaultErrorFallback} pendingFallback={<PageLoader />}>
              <TodoById />
            </AsyncBoundary>
          </div>
        </>
      )}
    </Container>
  );
};

export default TodoDetail;

const Container = styled.div`
  .todoByIdSection {
    width: 300px;
    height: 500px;
    position: fixed;
    right: 0;
    top: 50%;
    transform: translateY(-50%);
  }
`;
