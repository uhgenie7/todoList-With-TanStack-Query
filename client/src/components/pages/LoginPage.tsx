import AuthForm from '@src/components/organism/AuthForm';
import { useLoginQuery } from '@src/hooks/query/auth';
import ExtendsAsyncBoundary from '../boundaries/ExtendsAsyncBoundary';

const LoginPage = () => {
  return (
    <>
      <h2>로그인</h2>
      <div className="center">
        <ExtendsAsyncBoundary>
          <AuthForm useQuery={useLoginQuery} buttonValue="로그인" />
        </ExtendsAsyncBoundary>
      </div>
    </>
  );
};

export default LoginPage;
