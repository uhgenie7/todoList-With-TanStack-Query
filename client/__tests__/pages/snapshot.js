import { render } from '@testing-library/react';
import { Sample } from '@src/pages/Sample';

it('renders homepage unchanged', () => {
  const { container } = render(<Sample />);
  expect(container).toMatchSnapshot();
});
