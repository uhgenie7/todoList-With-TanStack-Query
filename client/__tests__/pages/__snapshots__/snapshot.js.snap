// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <main>
    <h1>
      test jest
    </h1>
    <p>
      ggg
    </p>
  </main>
</div>
`;
